addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+                )
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger()
+            else:
+                message = _(
+                    "%(matching_image_count)s matching images have been found for "
+                    "%(product_count)s products.",
+                    matching_image_count=len(self.products_to_process.filtered('image_1920')),
+                    product_count=len(self.products_to_process)
+                )
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'type': 'success',
+                    'title': _("Product images"),
+                    'message': message,
+                    'next': {'type': 'ir.actions.act_window_close'},
+                }
+            }
+
+    def _get_products_to_process(self, limit=10000):
+        """ Get the list of products that need to be processed
+
+        Use the wizard's list of products_to_process if it contains products, else search the db.
+
+        :param int limit: maximum number of records to return (default: 10000, which is the default
+        rate limit of Google's API)
+        :return: a list of products that meet the criteria
+        :rtype: list
+        """
+        products_to_process = self.products_to_process or self.env['product.product'].search([
+            ('image_fetch_pending', '=', 'True'),
+        ], limit=limit)
+        return products_to_process.filtered(
+            # p.image_fetch_pending needed for self.products_to_process's items that might have
+            # already been processed but not removed from the list (when called from
+            # action_fetch_image()).
+            lambda p: not p.image_1920 and p.barcode and p.image_fetch_pending
+        )[:limit]  # limit to apply after the filter when self.products_to_process is used.
+
+    def _cron_fetch_image(self):
+        """ Fetch images of a list of products using their barcode
+
+        This function is called from a cron job. If the daily request limit is reach, the cron job
+        will be schedule to run again a day later.
+        """
+        # Retrieve 100 products at a time to limit the cron's time length and avoid reach Google
+        # default rate limit.
+        products_to_process = self._get_products_to_process(100)
+        try:
+            self._fetch_image_from_google(products_to_process)
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                    fields.Datetime.now() + timedelta(days=1.0)
+                )
+            else:
+                raise
+        except ConnectionError:
+            self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                fields.Datetime.now() + timedelta(hours=1.0)
+            )
```

Comment:
This clause is used twice, and the body is the exact same in both cases. Maybe `_fetch_image_from_google` could just do that on its own iternally?